src/table/styles.js

```diff
@@ -0,0 +1,135 @@
+import { styled, Paper, TableContainer, TableRow, TableSortLabel, TableBody, Select, IconButton } from '@mui/material';
+
+// ---------- AnnounceWrapper ----------
+
+export const TableAnnouncer = styled('div')({
+  clip: 'rect(0 0 0 0)',
+  clipPath: 'inset(50%)',
+  height: '1px',
+  overflow: 'hidden',
+  position: 'absolute',
+  whiteSpace: 'nowrap',
+  width: '1px',
+});
+
+// ---------- FooterWrapper ----------
+
+export const StyledFooterWrapper = styled(Paper, {
+  shouldForwardProp: (prop) => prop !== 'tableTheme',
+})(({ tableTheme, theme }) => ({
+  height: 48,
+  display: 'flex',
+  justifyContent: 'flex-end',
+  alignItems: 'center',
+  paddingRight: theme.spacing(1),
+  boxShadow: 'none',
+  borderStyle: 'solid',
+  borderWidth: '0px 0px 1px 0px',
+  borderRadius: 0,
+  borderColor: tableTheme.pagination.borderColor,
+  color: tableTheme.pagination.color,
+  backgroundColor: tableTheme.backgroundColor,
+}));
+
+// ---------- PaginationContent ----------
+
+export const StyledSelect = styled(Select, {
+  shouldForwardProp: (prop) => prop !== 'paginationTheme',
+})(({ paginationTheme }) => ({
+  backgroundColor: 'inherit',
+  '& .MuiNativeSelect-icon': { color: paginationTheme.iconColor },
+}));
+
+export const StyledIconButton = styled(IconButton, {
+  shouldForwardProp: (prop) => prop !== 'disabledCondition' && prop !== 'paginationTheme',
+})(({ disabledCondition, paginationTheme }) => ({
+  color: disabledCondition ? paginationTheme.disabledIconColor : paginationTheme.iconColor,
+  cursor: disabledCondition ? 'default' : 'pointer',
+  height: '32px',
+}));
+
+// ---------- TableBodyWrapper ----------
+
+export const StyledTableBody = styled(TableBody, {
+  shouldForwardProp: (prop) => prop !== 'paginationNeeded' && prop !== 'bodyCellStyle',
+})(({ paginationNeeded, bodyCellStyle }) => ({
+  'tr :last-child': {
+    borderRight: paginationNeeded && 0,
+  },
+  'tr :first-child': {
+    borderLeft: !paginationNeeded && '1px solid rgb(217, 217, 217)',
+  },
+  '& td, th': {
+    fontSize: bodyCellStyle.fontSize,
+    padding: bodyCellStyle.padding,
+  },
+}));
+
+export const StyledBodyRow = styled(TableRow, {
+  shouldForwardProp: (prop) => prop !== 'bodyCellStyle',
```

Comment:
should be `shouldForwardProp: (prop) => prop !== 'hover && prop !== 'bodyCellStyle'',`?

Reply:
no hover is a property actually used by the component, so it should be passed